### PR TITLE
Feature/improve cloneability

### PR DESCRIPTION
**Pull request by pex-goldenc**
Touched up a few areas that were in the way of cloning externally.

### CODE REVIEW

File: src/PdfSharper/Pdf.AcroForms/PdfTextField.cs [hunk 1]
```diff
@@ -61,7 +61,9 @@ public PdfTextField(PdfDocument document, bool needsAppearance = false)
 
         public PdfTextField(PdfDictionary dict)
```
Comment (mcnallys) [inline src/PdfSharper/Pdf.AcroForms/PdfTextField.cs:62]: Do we need to pass the needs Appearances here?

File: src/PdfSharper/Pdf.AcroForms/PdfSignatureField.cs [hunk 2]
```diff
@@ -112,7 +110,6 @@ public PdfSignatureField(PdfDocument document)
             sign.Elements.Add(Keys.M, new PdfDate(DateTime.Now));
 
             document._irefTable.Add(sign);
-            document._irefTable.Add(this);
```
Comment (mcnallys) [inline src/PdfSharper/Pdf.AcroForms/PdfSignatureField.cs]: Do we know where this is getting added at? Our code or internally to pdf sharper?

Comment (pex-goldenc) [reply src/PdfSharper/Pdf.AcroForms/PdfSignatureField.cs]: PdfSignatureHandler.cs::AddSignatureComponents() is the only reference in the actual project. It grabs the Annotations array and adds this. Then it grabs the AcroForm's Fields and adds it to that as well. Do we need a separate add to the ireftable of the document? I thought those would take care of it. None of the other fields have explicit adding to the doc's table.

Comment (pex-goldenc) [inline src/PdfSharper/Pdf.AcroForms/PdfSignatureField.cs]: PdfSignatureHandler.cs::AddSignatureComponents() is the only reference in the actual project. It grabs the Annotations array and adds this. Then it grabs the AcroForm's Fields and adds it to that as well. Do we need a separate add to the ireftable of the document? I thought those would take care of it. None of the other fields have explicit adding to the doc's table.

File: src/PdfSharper/Pdf.AcroForms/PdfTextField.cs (hunk 1 above)
Comment (pex-goldenc) [reply src/PdfSharper/Pdf.AcroForms/PdfTextField.cs:62]: For consistency, yeah. Probably. Updating.

File: src/PdfSharper/Pdf.AcroForms/PdfSignatureField.cs (hunk 2 above)
Comment (mcnallys) [reply src/PdfSharper/Pdf.AcroForms/PdfSignatureField.cs]: The code here is adding an already signed signature field to the document, I think we should change it up to only add the field to the document;   (no explicit calls to ireftable) especially now that we have support for adding fields externally.

Signing the document should be a separate process.

**Review by mcnallys — APPROVED** [review]

**Pull request merged**